src/Pages/GlobalConfigurations/Authorization/PermissionGroups/List/PermissionGroupList.component.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useCallback, useMemo, useRef } from 'react'
+import {
+    SortingOrder,
+    SortableTableHeaderCell,
+    ErrorScreenNotAuthorized,
+    ERROR_EMPTY_SCREEN,
+    noop,
+    Pagination,
+    Reload,
+    TOAST_ACCESS_DENIED,
+    useAsync,
+    DEFAULT_BASE_PAGE_SIZE,
+    useUrlFilters,
+    abortPreviousRequests,
+    getIsRequestAborted,
+} from '@devtron-labs/devtron-fe-common-lib'
+import { API_STATUS_CODES } from '../../../../../config'
+
+import { getPermissionGroupList } from '../../authorization.service'
+import { permissionGroupLoading, SortableKeys } from './constants'
+import PermissionGroupListHeader from './PermissionGroupListHeader'
+import PermissionGroupRow from './PermissionGroupRow'
+import { useAuthorizationContext } from '../../AuthorizationProvider'
+import { importComponentFromFELibrary } from '../../../../../components/common'
+import FiltersEmptyState from '../../shared/components/FilterEmptyState/FilterEmptyState.component'
+import NoPermissionGroups from './NoPermissionGroups'
+
+const PermissionGroupInfoBar = importComponentFromFELibrary('PermissionGroupInfoBar', noop, 'function')
+
+const PermissionGroupList = () => {
+    const {
+        pageSize,
+        offset,
+        changePage,
+        changePageSize,
+        searchKey,
+        handleSearch,
+        sortBy,
+        handleSorting,
+        sortOrder,
+        clearFilters,
+    } = useUrlFilters<SortableKeys>({ initialSortKey: SortableKeys.name })
+    const filterConfig = useMemo(
+        () => ({
+            size: pageSize,
+            offset,
+            searchKey,
+            sortBy,
+            sortOrder,
+        }),
+        [pageSize, offset, searchKey, sortBy, sortOrder],
+    )
+    const abortControllerRef = useRef(new AbortController())
+    const [isLoading, result, error, reload] = useAsync(
+        () =>
+            abortPreviousRequests(
+                () => getPermissionGroupList(filterConfig, abortControllerRef.current.signal),
+                abortControllerRef,
+            ),
+        [filterConfig],
+    )
+    const { isAutoAssignFlowEnabled } = useAuthorizationContext()
+
+    const showLoadingState = isLoading || getIsRequestAborted(error)
+
+    const getPermissionGroupDataForExport = useCallback(
```

Comment:
We can either remove the useCallback from here or we should memoize the `< PermissionGroupListHeader ` component